### ISSUE TITLE
[BUG] Apply to all items overwrites item access settings even when "add to existing entries" option is selected

### ISSUE THREAD

**Issue by elynema** [avalonmediasystem/avalon#5227]
**Describe the bug**
When the user makes a change at the collection level (assign staff roles, item access, assign special access, item discovery), it only impacts new items unless it is applied to existing items. When the user chooses to apply to existing items, the collection's default item access level overwrites any custom item access level that may be set on items in the collection, even if the user chooses the option to "add to existing Special Access entries."

**To Reproduce**
Steps to reproduce the behavior, including the results:
1. Make sure that you have a collection with an item that has a different Item access setting than the default Item access setting for the collection.
2. Go to Manage Collection. 
3. Add a user under Avalon User
4. Select "apply to all existing items"
5. Select "Yes, add to existing Special Access entries" and click Apply
6. Go to Edit the item identified
7. In the Access Control section under Item access, you will see that the Item access setting has been changed to match the default setting for the collection.

**Expected behavior**
I would expect that the Item discovery and Item access default settings at the collection level would not overwrite those settings on individual items in the collection if the user selects "Yes, add to existing Special Access entries" in the popup.

**Environment (please complete the following information):**
 - Device / platform: Mac laptop
 - Browser: Chrome Version 113.0.5672.126
- [ ] avalon-dev
- [ ] avalon-staging
- [x] mco-staging
- [ ] mco-production
- [ ] demo

**Release:**
Problems were first reported by IULMIA in 7/2022, when we started logging data. Reported again on 6/5/2023.

**Additional notes**
This was only tested initially on mco-staging since the bug was reported in production. Should probably be tested 

# Done Looks Like
- [x] Controlled Digital Lending card moved out of Access Control content block
- [ ] "Apply to Existing" button added to each card block (item access, visibility, special access)
- [x] "Save" button added to each card block

# Mockups
Moving CDL card out of Access Control
![Screenshot 2023-06-20 at 8.58.23 AM.png](https://images.zenhubusercontent.com/5cc1c0106d0be2229caf2d9e/e2f38b98-76c8-4388-96fc-c2a2147a74ea)

Apply to Existing Items button added to individual cards for settings
![Screenshot 2023-06-20 at 9.10.41 AM.png](https://images.zenhubusercontent.com/5cc1c0106d0be2229caf2d9e/18b8cca5-4981-41f9-98e4-1e5d459bd50e)

**Comment by joncameron** [comment]
# 
- Move "Apply to Existing Items" to each permissions section (item access, visibility, special access)
- Large collections are difficult to update with Blacklight bookmarks
- # of items affected by the action should be in warning modal before action
- Save could also be moved to specific section
- CDL should be moved out of the item access section

**Comment by elynema** [comment]
@joncameron When a user selects "Apply to all existing items" does that also save the change at the collection level or does the Save Access Settings have to be clicked as well? When hitting "Apply to all existing items" do we have the same modal for choosing to replace or add onto existing entries? I am thinking that choice only makes sense in the Assign Special Access section, where there could be multiple values on the item that need to be retained?

**Comment by joncameron** [comment]
- I think the lending period could get collapsed into the "CDL" card if possible.
- "Save Setting" seems OK; other potential text: "Save This Setting", "Save Default Setting"
- Checkboxes/radio buttons with buttons below could be put into vertical alignment
- Should other actions have a modal as well, in addition to the "assign special access"?

**Comment by elynema** [comment]
Sounds like "assign special access" is the only section that needs to present the modal with a specific choice.
Other "apply to all existing items" buttons should tell the user via browser popup how many items are being changed?

**Comment by elynema** [comment]
@masaball The Assign Special Access section doesn't seem to have a "Save Setting" option. Is that intentional?

![image.png](https://images.zenhubusercontent.com/62d989a70c79cb5218735cd6/5d0d34f8-bb67-41fb-8f20-cd30199b34bc)

**Comment by masaball** [comment]
Yes, that is intentional. Each of the `Add` buttons is its own save, so a `Save Setting` for the whole special access form is redundant.

**Comment by elynema** [comment]
@joncameron Do we want to retain some type of Save and Continue button to be parallel with other edit item pages?

@joncameron  There is no Save button for the Assign Staff Roles section. Guessing the Add button does a save? As a user, I personally might not trust the Add button to actually perform the save, especially when some sections have a save button (it seems inconsistent). Not sure how others would feel.

@masaball The "Set Default Access Control for New Items" heading is important but also pretty subtle and really easy to miss. Looks like MCO has some whitespace above it that would be helpful.

Functionality seems to work as intended. I didn't test creating new items, just that the default values saved to the collection.

**Comment by masaball** [comment]
Good catch. What was happening is that the overwrite includes Read Groups which are part of determining visibility. So `media_object.visibility` was not being changed but the read group still got switched from 'private' to 'public' or vice versa, effectively causing the same result.

I'd caught a similar case for the non-overwrite where multiple visibilities were being assigned but didn't catch that it would have the same type of problem when overwriting. I'll get a fix PR up shortly.

**Comment by elynema** [comment]
@masaball Jon I discussed today and agreed to leave the Save Settings buttons as they are for now and we can demo this to the MCO Working Group in a couple of weeks to see if it all makes sense to them.

I'd still like to see a little whitespace above the "Set Default Access Control for New Items" heading on the collections page.

Tested changing collection settings and creating a new item and worked properly, so I think the whitespace is the last change.
